sdk/identity/identity/src/credentials/azurePipelinesServiceConnectionCredential.ts

```diff
@@ -0,0 +1,168 @@
+   * @param options - The options used to configure any requests this
+   *                TokenCredential implementation might make.
+   */
+  public async getToken(
+    scopes: string | string[],
+    options?: GetTokenOptions
+  ): Promise<AccessToken> {
+    if (!this.clientAssertionCredential) {
+      const errorMessage = `${credentialName}: is unavailable. tenantId, clientId, and either serviceConnectionId are required parameters. 
+      To use Federation Identity in Azure Pipelines, these are required as inputs / env variables - 
+      tenantId,
+      clientId,
+      serviceConnectionId,
+      "SYSTEM_TEAMFOUNDATIONCOLLECTIONURI" &&
+      "SYSTEM_TEAMPROJECTID" &&
+      "SYSTEM_PLANID" &&
+      "SYSTEM_JOBID" &&
+      "SYSTEM_ACCESSTOKEN"
+      See the troubleshooting guide for more information: https://aka.ms/azsdk/js/identity/troubleshoot`;
+      logger.error(errorMessage);
+      throw new CredentialUnavailableError(errorMessage);
+    }
+    logger.info("Invoking getToken() of Client Assertion Credential");
+    return this.clientAssertionCredential.getToken(scopes, options);
+  }
+
+  /**
+   *
+   * @param oidcRequestUrl - oidc request url
+   * @param systemAccessToken - system access token
+   * @returns OIDC token from Azure Pipelines
+   */
+  private async requestOidcToken(
+    oidcRequestUrl: string,
+    systemAccessToken: string
+  ): Promise<string> {
+    logger.info("Requesting OIDC token from Azure Pipelines...");
+    logger.info(oidcRequestUrl);
+
+    const httpClient = createDefaultHttpClient();
+
+    const request = createPipelineRequest({
+      url: oidcRequestUrl,
+      method: "POST",
+      headers: createHttpHeaders({
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${systemAccessToken}`,
+      }),
+    });
+
+    const response = await httpClient.sendRequest(request);
+    const text = response.bodyAsText;
+    if (!text) {
+      throw new AuthenticationError(
+        response.status,
+        `${credentialName}: Authenticated Failed. Received null token from OIDC request.`
+      );
+    }
+    const result = JSON.parse(text);
+    if (result?.oidcToken) {
+      return result.oidcToken;
+    } else {
+      throw new AuthenticationError(
+        response.status,
+        `${credentialName}: Authentication Failed. oidcToken field not detected in the response. Response = ${JSON.stringify(
+          result
+        )}`
+      );
+    }
+  }
+
+  /**
+   * Ensures all system env vars are there to form the request uri for OIDC token
+   * @returns void
+   * @throws CredentialUnavailableError
+   */
+  private ensurePipelinesSystemVars(): void {
+    if (
+      process.env.SYSTEM_TEAMFOUNDATIONCOLLECTIONURI &&
+      process.env.SYSTEM_TEAMPROJECTID &&
+      process.env.SYSTEM_PLANID &&
+      process.env.SYSTEM_JOBID &&
+      process.env.SYSTEM_ACCESSTOKEN
+    ) {
+      return;
+    }
+    const missingEnvVars = [];
```

Comment:
nit (feel free to ignore): a loop would be easier to follow and modify in the future

Pseudocode:
```
const requiredEnvVars = ["SYSTEM_TEAMFOUNDATIONCOLLECTIONURI", "SYSTEM_TEAMPROJECTID" ....etc]
const missingEnvVars = requiredEnvVars.filter(e => !process.env[e])
if (missingEnvVars.length > 0) { etc
```